zebra/zebra_dplane.c

```diff
@@ -280,34 +280,8 @@ struct dplane_neigh_table {
  * Policy based routing rule info for the dataplane
  */
 struct dplane_ctx_rule {
-	uint32_t seq;
-	uint32_t priority;
-	uint32_t unique;
+	struct pbr_rule prule;
 
-	/* The route table pointed by this rule */
-	uint32_t table;
-
-	/* Filter criteria */
-	uint32_t filter_bm;
-	uint32_t fwmark;
-	uint8_t dsfield;
-	struct prefix src_ip;
-	struct prefix dst_ip;
-	uint8_t ip_proto;
-	uint16_t src_port;
-	uint16_t dst_port;
-
-	uint8_t action_pcp;
-	uint16_t action_vlan_id;
-	uint16_t action_vlan_flags;
-
-	uint32_t action_queue_id;
-
-	uint8_t filter_pcp;
-	uint16_t filter_vlan_id;
-	uint16_t filter_vlan_flags;
-
-	char ifname[INTERFACE_NAMSIZ + 1];
 	struct ethaddr smac;
```

Comment:
are these other attributes not part of the pbr struct - should they be converted too?

Reply:
The other attributes (smac, dmac, out_ifindex, and dp_flow_ptr) are not part of the lib/pbr.h struct pbr_rule - instead, they are derived from state in zebra.